x/ecocredit/server/location_test.go

```diff
@@ -0,0 +1,79 @@
+package server
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateLocation(t *testing.T) {
+	specs := []struct {
+		name      string
+		location  string
+		expectErr bool
+	}{
+		{
+			name:      "can validate valid country code",
+			location:  "AF",
+			expectErr: false,
+		},
+		{
+			name:      "can validate valid country code 2",
+			location:  "BF",
+			expectErr: false,
+		},
+		{
+			name:      "can't validate invalid country code",
+			location:  "ZZ",
+			expectErr: true,
+		},
+		{
+			name:      "can't validate invalid country code 2",
+			location:  "Z!adflksdfZ",
+			expectErr: true,
+		},
+		{
+			name:      "can validate valid region code",
+			location:  "AF-BDS",
+			expectErr: false,
+		},
+		{
+			name:      "can validate valid region code 2",
+			location:  "BF-BAL",
+			expectErr: false,
+		},
+		{
+			name:      "can't validate invalid region code",
+			location:  "BF-ZZ",
+			expectErr: true,
+		},
+		{
+			name:      "can validate valid postal code",
+			location:  "BF-BAL-1",
```

Comment:
I would say we want a space instead of a `-` before the postal code. What do you think?

Reply:
Not sure! Why would you prefer one to the other?

Reply:
Hmm just seems more visually clear

Reply:
Sure, I can just work the whole thing into a single regex then